File: week_02/src/main/java/Question11.java
/*
Autor  = Zaid Nayeb
Date   = 17/11/2025

11.	Simple Calculator using switch-case
Description: Input two numbers and an operator (+, -, *, /) and compute the result.
Example: Input:
5, 3, +
Output:
8

 */
public class Question11 {
    public static double calculator(double a, double b, char c){
        double result = 0;
        switch (c){
            case '+':
                result = a + b;
                break;
            case '-':
                result = a - b;
                break;
            case '*':
                result = a * b;
                break;
            case '/':
                result = a / b;
                break;
        }
        return result;
    }
}

File: week_03/src/Question_11.js
// Author : Zaid Nayeb
// Date   : 2025/11/26
/*
11. Generate all prime numbers up to N 
Description: List all prime numbers up to a given limit using a loop. 
Example: 
Input: 
10 Output: 
2, 3, 5, 7 

*/
function isPrime(num) {
    if (num < 2) {
        return false;
    }

    for (let i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) {
            return false;
        }
    }
    return true;
}


function primeNumbers(number) {
    let primes = [];
    for (let i = 0; i <= number; i++) {
        if (isPrime(i)) {
            primes.push(i);
        }
    }
    return primes;
}
console.log(primeNumbers(10));

module.exports = { isPrime, primeNumbers };
